src/NodaTime/CalendarSystem.cs

```diff
@@ -634,6 +634,21 @@ public int GetMonthsInYear(int year)
             return YearMonthDayCalculator.GetMonthsInYear(year);
         }
 
+        /// <summary>
+        /// Returns the number of weeks in the given week-year.
+        /// </summary>
+        /// <param name="weekYear">The week-year to determine the number of weeks for.</param>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="weekYear"/> is outside
+        /// the range of this calendar. </exception>
+        /// <returns>The number of weeks in the given week-year.</returns>
+        public int GetWeeksInWeekYear(int weekYear)
+        {
+            // TODO: Is this really valid? It means we can't necessarily round-trip. Maybe be more lenient, but then check the resulting days-since-epoch.
```

Comment:
Will add a test when I start addressing the issue. Basically, a date can be in a different year to its week-year... so while it would be nice to have:

```csharp
LocalDate date = ...;
LocalDate date2 = LocalDate.FromWeekYearWeekAndDay(date.WeekYear, date.WeekofWeekYear, date.IsoDayOfWeek);
```

... work on all dates, I suspect it may fail for extremes where the week-year may be beyond the maximum year or below the minimum year.

(We also need an overload accepting a CalendarSystem, as currently it always assumes the ISO calendar...)